internal/rsl/rsl.go

```diff
@@ -411,6 +423,30 @@ func GetLatestReferenceEntryForRefBefore(repo *git.Repository, refName string, a
 	return targetEntry, annotations, nil
 }
 
+// GetLatestUnskippedReferenceEntryForRef returns the latest reference entry for
+// the ref that does not have an annotation marking it as to-be-skipped.
+func GetLatestUnskippedReferenceEntryForRef(repo *git.Repository, refName string) (*ReferenceEntry, []*AnnotationEntry, error) {
+	return GetLatestUnskippedReferenceEntryForRefBefore(repo, refName, plumbing.ZeroHash)
+}
+
+// GetLatestUnskippedReferenceEntryForRefBefore returns the first reference
+// entry for the ref before the anchor that does not have an annotation marking
+// it as to-be-skipped.
+func GetLatestUnskippedReferenceEntryForRefBefore(repo *git.Repository, refName string, anchor plumbing.Hash) (*ReferenceEntry, []*AnnotationEntry, error) {
```

Comment:
It can be ambiguous what before means because you can interpret it as chronologically before if walking from the start of the repo, but this means iteratively before walking the repo starting from the ref.

Maybe clarify with this?

```suggestion
// GetLatestUnskippedReferenceEntryForRefBefore returns the first reference
// entry for the ref that does not have an annotation marking it as to-be-skipped.
// Entries are searched starting from the ref until the anchor is reached.
func GetLatestUnskippedReferenceEntryForRefBefore(repo *git.Repository, refName string, anchor plumbing.Hash) (*ReferenceEntry, []*AnnotationEntry, error) {
```

Reply:
I think I'm a little confused by the distinction between chronologically and iterative. Could you clarify? Thanks!